### PR TITLE
Forum Post Response [Part 2]: Add text answer in submission page

### PR DESCRIPTION
**Pull request by aylz**
Context: #4060 

The feature to allow forum posts to be submitted in an assessment would be implemented over multiple PRs.

This PR enables text answers to be entered and viewed in submissions.

Part 1: #4102 Allow CRUD from assessment page

### CODE REVIEW

File: app/models/course/assessment/question/forum_post_response.rb [hunk 1]
```diff
@@ -9,6 +9,20 @@ def question_type
     I18n.t('course.assessment.question.forum_post_responses.question_type')
   end
 
+  def attempt(submission, last_attempt = nil)
+    answer =
+      Course::Assessment::Answer::ForumPostResponse.new(submission: submission, question: question)
+    last_attempt&.answer_options&.each do |answer_option|
```
Comment (zhuhanming) [inline app/models/course/assessment/question/forum_post_response.rb:15]: Can I check what `answer_options` here is?

File: app/views/course/assessment/answer/forum_post_responses/_forum_post_response.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+json.fields do
+  json.questionId answer.question_id
+  json.id answer.acting_as.id
+  json.answer_text answer.answer_text
```
Comment (zhuhanming) [inline app/views/course/assessment/answer/forum_post_responses/_forum_post_response.json.jbuilder:5]: Can we standardise? Either `camelCase` or `snake_case`. I would recommend going with `json.answerText`.

File: app/views/course/assessment/question/forum_post_responses/_forum_post_response.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+json.autogradable question.auto_gradable?
+json.has_text_response question.has_text_response
```
Comment (zhuhanming) [inline app/views/course/assessment/question/forum_post_responses/_forum_post_response.json.jbuilder:3]: Ditto

**Review by zhuhanming — APPROVED** [review]
I think generally looks good. See comments below.

Also, do rebase this and Part 1 both.

File: app/views/course/assessment/answer/forum_post_responses/_forum_post_response.json.jbuilder (hunk 2 above)
Comment (aylz) [reply app/views/course/assessment/answer/forum_post_responses/_forum_post_response.json.jbuilder:5]: Same thoughts on the standardisation, but other question types seem to have a blend of `camelCase` and `snake_case`. The former is typically used for `questionId`, and the latter for other variables specific to the question type.

File: app/models/course/assessment/question/forum_post_response.rb (hunk 1 above)
Comment (aylz) [reply app/models/course/assessment/question/forum_post_response.rb:15]: will be fixing this in Part 4

**Pull request closed without merging**